code/game/objects/structures/electricchair.dm

```diff
@@ -1,52 +1,132 @@
+	if(guinea_pig)
+		UnregisterSignal(guinea_pig, list(COMSIG_PARENT_PREQDELETED))
+	parent_chair = null
+	used_shock_kit = null
+	guinea_pig  = null
+	STOP_PROCESSING(SSprocessing, src)
+
+/datum/component/electrified_chair/proc/delete_self()
+	SIGNAL_HANDLER
+	qdel(src)
+
+///checks if the shock kit connected to parent is still there and unregisters if it isnt
+/datum/component/electrified_chair/proc/check_shock_kit(datum/source, atom/movable/AM, atom/newLoc)
+	SIGNAL_HANDLER
+	if(used_shock_kit == AM && newLoc != parent_chair)
+		delete_self()
+
+/datum/component/electrified_chair/proc/on_buckle(datum/source, mob/living/mob_to_buckle, _force)
+	SIGNAL_HANDLER
+	if(!istype(mob_to_buckle) || guinea_pig)
+		return FALSE
+	guinea_pig = mob_to_buckle
+	RegisterSignal(guinea_pig, COMSIG_PARENT_PREQDELETED, .proc/nullify_guinea_pig)
+	RegisterSignal(parent_chair, COMSIG_MOVABLE_UNBUCKLE, .proc/nullify_guinea_pig)
+	START_PROCESSING(SSprocessing, src)
+	return TRUE
+
+///for whatever reason the guinea pig is gone so we cant shock them anymore
+/datum/component/electrified_chair/proc/nullify_guinea_pig()
+	SIGNAL_HANDLER
+	UnregisterSignal(guinea_pig, COMSIG_PARENT_PREQDELETED)
+	UnregisterSignal(parent_chair, COMSIG_MOVABLE_UNBUCKLE)
+	guinea_pig = null
+	STOP_PROCESSING(SSprocessing, src)
+
+///where the guinea pig is actually shocked if possible
+/datum/component/electrified_chair/process(delta_time)
+	if(QDELETED(guinea_pig) || QDELETED(used_shock_kit) || QDELETED(parent_chair))
+		time_since_last_shock = 0
+		return PROCESS_KILL
+
+	if(time_since_last_shock < ELECTRIC_CHAIR_WAIT_TIME)
+		time_since_last_shock += delta_time
 		return
-	A.use_power(AREA_USAGE_EQUIP, 5000)
-
-	flick("echair_shock", src)
-	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
-	s.set_up(12, 1, src)
-	s.start()
-	if(has_buckled_mobs())
-		for(var/m in buckled_mobs)
-			var/mob/living/buckled_mob = m
-			buckled_mob.electrocute_act(85, src, 1)
-			to_chat(buckled_mob, "<span class='userdanger'>You feel a deep shock course through your body!</span>")
-			addtimer(CALLBACK(buckled_mob, /mob/living.proc/electrocute_act, 85, src, 1), 1)
-	visible_message("<span class='danger'>The electric chair went off!</span>", "<span class='hear'>You hear a deep sharp shock!</span>")
-
-/obj/structure/chair/e_chair/post_buckle_mob(mob/living/L)
-	SEND_SIGNAL(L, COMSIG_ADD_MOOD_EVENT, "dying", /datum/mood_event/deaths_door)
-
-/obj/structure/chair/e_chair/post_unbuckle_mob(mob/living/L)
-	SEND_SIGNAL(L, COMSIG_CLEAR_MOOD_EVENT, "dying")
+	time_since_last_shock = 0
+
+	var/turf/our_turf = get_turf(parent_chair)
+	var/obj/structure/cable/live_cable = our_turf.get_cable_node()
+
+	if(parent_chair.has_buckled_mobs() && guinea_pig in parent_chair.buckled_mobs)
+		if(!live_cable || !live_cable.powernet || live_cable.powernet.avail < 10)
+			return
+		var/shock_damage = round(live_cable.powernet.avail/5000)
```

Comment:
Put the 5000 in a define or ain a variable.

Reply:
done